codegenerator/cli/templates/static/codegen/src/eventFetching/rpc/RpcSource.res

```diff
@@ -60,28 +112,49 @@ let getSuggestedBlockIntervalFromExn = {
           | _ => None
           }
         | None =>
+          // Try each provider's specific error pattern
           switch blockRangeLimitRegExp->Js.Re.exec_(message) {
-          | Some(execResult) =>
-            switch execResult->Js.Re.captures {
-            | [_, Js.Nullable.Value(blockRangeLimit)] =>
-              switch blockRangeLimit->Int.fromString {
-              | Some(blockRangeLimit) if blockRangeLimit > 0 => Some(blockRangeLimit)
-              | _ => None
-              }
-            | _ => None
-            }
+          | Some(execResult) => extractBlockRange(execResult)
           | None =>
             switch alchemyRangeRegExp->Js.Re.exec_(message) {
-            | Some(execResult) =>
-              switch execResult->Js.Re.captures {
-              | [_, Js.Nullable.Value(blockRangeLimit)] =>
-                switch blockRangeLimit->Int.fromString {
-                | Some(blockRangeLimit) if blockRangeLimit > 0 => Some(blockRangeLimit)
-                | _ => None
+            | Some(execResult) => extractBlockRange(execResult)
+            | None =>
+              switch cloudflareRangeRegExp->Js.Re.exec_(message) {
+              | Some(execResult) => extractBlockRange(execResult)
+              | None =>
+                switch thirdwebRangeRegExp->Js.Re.exec_(message) {
+                | Some(execResult) => extractBlockRange(execResult)
+                | None =>
+                  switch blockpiRangeRegExp->Js.Re.exec_(message) {
+                  | Some(execResult) => extractBlockRange(execResult)
+                  | None =>
+                    switch baseRangeRegExp->Js.Re.exec_(message) {
+                    | Some(_) => Some(2000)
+                    | None =>
+                      switch blastPaidRegExp->Js.Re.exec_(message) {
+                      | Some(execResult) => extractBlockRange(execResult)
+                      | None =>
+                        switch chainstackRegExp->Js.Re.exec_(message) {
+                        | Some(_) => Some(10000)
+                        | None =>
+                          switch coinbaseRegExp->Js.Re.exec_(message) {
+                          | Some(execResult) => extractBlockRange(execResult)
+                          | None =>
+                            switch publicNodeRegExp->Js.Re.exec_(message) {
+                            | Some(execResult) => extractBlockRange(execResult)
+                            | None =>
+                              switch hyperliquidRegExp->Js.Re.exec_(message) {
+                              | Some(execResult) => extractBlockRange(execResult)
+                              | None => None
```

Comment:
Would this not look cleaner with a more flat structure of chaining Option.mapOr together?

Can even make a `tryExec` function that  is reused eg `tryExec(alchemyRangeRegExp)` or similar. But this is fine too :+1: